### PR TITLE
Make create_cluster sync

### LINKED ISSUE #30: `create_cluster` returning a coroutine not a `ClusterManager`
Following your example in dask/dask-jobqueue#543 I'm trying to create a cluster using `create_cluster()`

```yaml
version: 1
module: "dask_jobqueue"
class: "SLURMCluster"
args: []
kwargs:
  name: 'dask-worker'
  cores: 112
  memory: '750 GB'
  processes: 14
  queue: 'cpu'
  walltime: '00:30:00'
```

```python
import dask_ctl

cluster = dask_ctl.create_cluster('cluster_spec.yaml')
```

The returned object is a coroutine, not a cluster manager - is this expected? 

I went down the rabbit hole of awaiting it in an async function but wanted to check this was necessary, the docs seem to imply it shouldn't be

Sorry for the barrage of extra work here!

### PR DESCRIPTION
**Pull request by jacobtomlinson**
Closes #30.

Everything in `dask_ctl.lifecycle` is an async function wrapped in a sync function except for `create_cluster`. Make it consistent.

### CODE REVIEW

**Review by alisterburt — APPROVED** [review]
works in my hands! 🥳

**Comment by alisterburt** [comment]
I took a look at the test failures but the error message seems not to be useful - there is an asyncio plugin in the test deps which installs correctly on CI at the pinned version...

As an aside, once this goes in would you be okay to cut a minor release? :)

**Comment by jacobtomlinson** [comment]
The test seems to be unable to find the running cluster, but it also seems to be failing intermittently. I've added a couple of lines to try and give the cluster more time to get started in case this is caused by a race condition.

File: dask_ctl/tests/test_lifecycle.py [hunk 1]
```diff
@@ -7,13 +7,15 @@
...
-    cluster = await create_cluster(simple_spec_path)
+def test_create_cluster(simple_spec_path):
+    cluster = create_cluster(simple_spec_path)
 
     assert isinstance(cluster, LocalCluster)
 
 
+@pytest.mark.xfail(
+    "GitHib Actions seems to intermittently have port 8786 in use causing this test to fail."
+)
```
Comment (keewis) [inline dask_ctl/tests/test_lifecycle.py:18]:
```suggestion
@pytest.mark.xfail(
    reason="GitHib Actions seems to intermittently have port 8786 in use causing this test to fail."
)
```

**Pull request merged**